--enable-mpi1-compatibility broken on v4.0.x

When --enable-mpi1-compatibility is added to the configure line below, I receive an error at `make install` time:

```
...
Making install in mpi/fortran/use-mpi-tkr
make[2]: Entering directory `/tmp/gpaulsen/ompi/ompi/mpi/fortran/use-mpi-tkr'
  PPFC     mpi.lo
pmpi-f90-removed-interfaces.h:115.53:
    Included at mpi.F90:69:

subroutine PMPI_Type_extent(datatype, extent, ierror)
                                                     1
Error: Symbol 'pmpi_type_extent' at (1) already has an explicit interface
pmpi-f90-removed-interfaces.h:116.33:
    Included at mpi.F90:69:

  integer, intent(in) :: datatype
                                 1
Error: Unexpected data declaration statement in INTERFACE block at (1)
pmpi-f90-removed-interfaces.h:117.32:
    Included at mpi.F90:69:

  integer, intent(out) :: extent
                                1
Error: Unexpected data declaration statement in INTERFACE block at (1)
pmpi-f90-removed-interfaces.h:118.32:
    Included at mpi.F90:69:

  integer, intent(out) :: ierror
                                1
Error: Unexpected data declaration statement in INTERFACE block at (1)
pmpi-f90-removed-interfaces.h:119.3:
    Included at mpi.F90:69:

end subroutine PMPI_Type_extent
   1
Error: Expecting END INTERFACE statement at (1)
make[2]: *** [mpi.lo] Error 1
make[2]: Leaving directory `/tmp/gpaulsen/ompi/ompi/mpi/fortran/use-mpi-tkr'
make[1]: *** [install-recursive] Error 1
make[1]: Leaving directory `/tmp/gpaulsen/ompi/ompi'
make: *** [install-recursive] Error 1
```

### What version of Open MPI are you using? (e.g., v1.10.3, v2.1.0, git branch name and hash, etc.)
v4.0.x

### Describe how Open MPI was installed (e.g., from a source/distribution tarball, from a git clone, from an operating system distribution package, etc.)

* Cloned from github, autogen.pl and then configured.

### Please describe the system on which you are running
* POWER8E rev: 2.1 (pvr 004b 0201)
* `uname -r`: 3.10.0-514.el7.ppc64le
* Red Hat Enterprise Linux Server release 7.3
* gcc (GCC) 4.8.5 20150623 (Red Hat 4.8.5-11)
* GNU Fortran (GCC) 4.8.5 20150623 (Red Hat 4.8.5-11)

### Config.log
* [config.log](https://gist.github.com/gpaulsen/61c3568673679b344449e8edb44a642d)

--- a/ompi/mpi/fortran/use-mpi-tkr/pmpi-f90-interfaces.h
+++ b/ompi/mpi/fortran/use-mpi-tkr/pmpi-f90-interfaces.h
@@ -1595,17 +1595,6 @@ end subroutine PMPI_Type_dup
 end interface
 
 
-interface PMPI_Type_extent
-
-subroutine PMPI_Type_extent(datatype, extent, ierror)
-  integer, intent(in) :: datatype
-  integer, intent(out) :: extent
-  integer, intent(out) :: ierror
-end subroutine PMPI_Type_extent
-
-end interface
-
-
 interface PMPI_Type_free
 
 subroutine PMPI_Type_free(datatype, ierror)

